### DISCUSSION TITLE
Cannot import typescript module with generics from outside the project

### DISCUSSION THREAD

**Discussion by defr0std** [vercel/next.js#36414] [Help]
Hi,

We would like to re-use some of the existing typescript modules in our project, which are located outside of the next.js directory.

Imagine the following structure:
```
project/
  outside-module.ts
  nextjs/
    pages/
      index.tsx
```    

If I try to import a function with a _generic type_ from outside of the `nextjs` directory, I am getting the following error:

```typescript
// In file outside-module.ts:
export function OutsideProject<T>() {}
```

```
error - ../outside-module.ts
Module parse failed: Unexpected token (1:30)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> export function OutsideProject<T>() {
| }
|
```

Inside the `nextjs` directory, everything works fine.

Here is an example on github: https://github.com/dennisfrostlander/next-js-generic:
* [Here](https://github.com/dennisfrostlander/next-js-generic/blob/main/my-app/pages/index.tsx#L8-L9) we call a generic function from inside and outside the project dir.
* The error is reported only for the module from outside the project dir.

**Comment by vadymshymko** [comment]
Hi. I recommend to look at [this issue](https://github.com/vercel/next.js/issues/9474). More specifically, [this comment](https://github.com/vercel/next.js/issues/9474#issuecomment-810212174). It looks like we are already able to solve this problem via next.config.js)
